public/app/features/manage-dashboards/state/RecentlyDeletedActions.tsx

```diff
@@ -29,10 +29,26 @@ export function RecentlyDeletedActions() {
   };
 
   const onRestore = async () => {
-    const promises = Object.entries(selectedItems.dashboard)
+    const resultsView = stateManager.state.result?.view.toArray();
+    if (!resultsView) {
+      return;
+    }
+
+    const uids = Object.entries(selectedItems.dashboard)
       .filter(([_, selected]) => selected)
-      .map(([uid]) => restoreDashboard({ dashboardUID: uid }));
+      .map((v) => v[0]);
+
+    const promises = uids.map((uid) => {
+      return restoreDashboard({ dashboardUID: uid });
+    });
+
     await Promise.all(promises);
+
+    const parentUIDs = uids
+      .map((uid) => resultsView.find((v) => v.uid === uid)?.location)
+      .filter((uid, index, self) => self.indexOf(uid) === index);
```

Comment:
really minor nits:
- we're handling `undefined`/`GENERAL_FOLDER_UID` the same in `clearFolders`, but we don't filter them properly for uniqueness here
- would this be clearer with a set? e.g. 
```tsx
const parentUIDs = new Set();
for (const uid of uids) {
  const parentUID = resultsView.find((v) => v.uid === uid)?.location) ?? GENERAL_FOLDER_UID;
  parentUIDs.add(parentUID);
}
dispatch(clearFolders(Array.from(parentUIDs)));
```
then we get proper uniqueness, and the type for `clearFolders` can just be `action: PayloadAction<string>` with no pesky `undefined`s floating around

Reply:
oh i see that was your first commit 😂 yeah basically